version/version.go

```diff
@@ -60,53 +65,39 @@ func Semver() string {
 	return version + "#" + gitSHA
 }
 
-// BuildTypeFromTag discovers the BuildType of the git tag.
-func BuildTypeFromTag(tag string) BuildType {
-	if tag == "" {
-		return Nightly
+// BuildEnv provides methods for determining version info from the current env.
+type BuildEnv interface {
+	IsCI() bool
+	IsNightly() bool
+	GetMostRecentTag() string
+}
+
+// ParseBuildEnv discovers the most recent tag and BuildType using the current
+// build environment
+func ParseBuildEnv(env BuildEnv) (string, BuildType) {
```

Comment:
This is kind of the opposite of parsing, you're taking a data structure and returning string data.

maybe `FindBuildType` or something?

Reply:
There is some string processing in the function, but overall you're right it's not really a parser. Will update naming to match.